### PR TITLE
Ls 22

### PR DESCRIPTION
**Pull request by IgorOksentyuk**
Login logic and styles fix.

### CODE REVIEW

File: src/app/administration/modal/modal.component.ts [hunk 1]
```diff
@@ -44,11 +40,24 @@ export class ModalComponent {
   }
 
   ok() {
-    this.modalService.configuration$.subscribe((productConfig) => {
-      productConfig.id = this.productId;
-      this.productAddEvent.emit(productConfig);
+    this.subscription = this.modalService.configuration$.subscribe(
+      (productConfig) => {
+        this.productsService.create(productConfig).subscribe(
+          () => {
```
Comment (alexsednev) [inline src/app/administration/modal/modal.component.ts]: Subscription in subscription. Use switchMap/mergeMap instead

File: src/app/administration/edit-modal/edit-modal.component.ts [hunk 2]
```diff
@@ -76,10 +79,24 @@ export class EditModalComponent {
   }
 
   ok() {
-    this.modalService.configuration$.subscribe((productConfig) => {
-      this.productEditvent.emit(productConfig);
+    this.subscription = this.modalService.configuration$.subscribe(
+      (productConfig) => {
+        this.productsService.update(productConfig).subscribe(
```
Comment (alexsednev) [inline src/app/administration/edit-modal/edit-modal.component.ts]: Use mergeMap or switchMap instead subscription in subscription

File: src/app/administration/modal/modal.component.ts [hunk 3]
```diff
@@ -44,11 +40,24 @@ export class ModalComponent {
...
+            this.productsService.newCreateEvent(productConfig);
+            this.dialogRef.close();
+          },
+          (badResponse) => {
+            this.errors = badResponse.error.message;
```
Comment (alexsednev) [inline src/app/administration/modal/modal.component.ts]: I assume that errors should be iterable value. (<div *ngFor="let error of errors">{{ error }}</div>). Are you sure that badResponse.error.message is an array?

File: src/app/administration/products/products.component.ts [hunk 4]
```diff
@@ -107,62 +125,48 @@ export class ProductsComponent implements OnInit {
...
   deleteProduct(id: string) {
     this.dialogRef
       .open(DeleteModalComponent, { width: '570px', height: '413px' })
       .afterClosed()
-      .subscribe(() => {
-        console.log(id);
-        if (id) {
+      .subscribe((res) => {
+        if (res) {
           this.productsService.delete(id).subscribe((res) => {
```
Comment (alexsednev) [inline src/app/administration/products/products.component.ts]: Same goes here. Use mergeMap. Or switchMap

File: src/app/login/login.component.html [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+
+      <div
+        class="alert alert-danger"
+        role="alert"
+        *ngIf="password.errors && f.submitted"
+      >
+        <div *ngIf="password.errors['required']">Password is required</div>
+        <div *ngIf="password.errors['minlength']">
+          Password must be at least 6 characters
+        </div>
```
Comment (alexsednev) [inline src/app/login/login.component.html]: Here you can use angular directive ngFor and don't hardcode each error.

File: src/app/login/login.component.ts [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+@Component({
+  selector: 'app-login',
+  templateUrl: './login.component.html',
+  styleUrls: ['./login.component.scss'],
+})
+export class LoginComponent {
+  form: any = {
+    username: null,
+    password: null,
+  };
```
Comment (alexsednev) [inline src/app/login/login.component.ts]: Please create type or interface and don't use 'any'. Good practice is avoid using any

**Review by alexsednev — CHANGES_REQUESTED** [review]

**Comment by alexsednev** [comment]
Also according to required criteria you should use reactive forms in you app

File: src/app/login/login.component.ts (hunk 6 above)
Comment (IgorOksentyuk) [reply src/app/login/login.component.ts]: Done.

File: src/app/administration/modal/modal.component.ts (hunk 3 above)
Comment (IgorOksentyuk) [reply src/app/administration/modal/modal.component.ts]: Done.

File: src/app/administration/modal/modal.component.ts (hunk 1 above)
Comment (IgorOksentyuk) [reply src/app/administration/modal/modal.component.ts]: Done.

File: src/app/administration/products/products.component.ts (hunk 4 above)
Comment (IgorOksentyuk) [reply src/app/administration/products/products.component.ts]: Done

File: src/app/administration/edit-modal/edit-modal.component.ts (hunk 2 above)
Comment (IgorOksentyuk) [reply src/app/administration/edit-modal/edit-modal.component.ts]: Done.

File: src/app/login/login.component.html (hunk 5 above)
Comment (IgorOksentyuk) [reply src/app/login/login.component.html]: Done.

**Pull request merged**